proof/refine/InvariantsPre_H.thy

```diff
@@ -5,45 +5,76 @@
  * SPDX-License-Identifier: GPL-2.0-only
  *)
 
-theory Invariants_H
+(* Preliminary definitions required to define arch-specific invariants and properties.
+   These are primarily related to obj_at' and storability.
+*)
+
+theory InvariantsPre_H
 imports
   LevityCatch
   "AInvs.ArchDetSchedSchedule_AI"
   "Lib.Heap_List"
+  Move_R
 begin
 
 (* global data and code of the kernel, not covered by any cap *)
 axiomatization
   kernel_data_refs :: "word64 set"
 
-context Arch begin
 
-declare lookupPTSlotFromLevel.simps[simp del]
-declare lookupPTFromLevel.simps[simp del]
+section \<open>Locale Setup for Idempotence over kernel_state Field Updates\<close>
```

Comment:
Very nice job collecting and rationalising these locales!

Just trying to think of what might help us in the future, do we want a comment here with a quick explanation of what we mean by idempotence over field updates, and maybe with an explanation that the locales are actually populated and interpreted in Invariants_H? It would be harder to explain here, but it might also help to have a comment explaining that completing the lattice is helpful to avoid future 'landmines', even if some of the locales aren't used now.

More ambitious ideas that almost definitely aren't worth the time would be to have some sort of test that confirms that the lattice is complete, and to have a new tool that constructs the lattice given `Arch` and the base locales.

Reply:
comments added

Reply:
I'm happy with the new comments.

Reply:
Same